test/unit/routes.test.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const sinon = require('sinon')
+
+const paths = require('../../app/paths')
+const routes = require('../../app/routes')
+const oldAuthorisationMiddleware = require('../../app/services/auth.service').enforceUserAuthenticated
+
+const pathsNotRequiringAuthentication = [
+  '/style-guide',
+  paths.user.logIn,
+  paths.user.otpLogIn,
+  paths.user.otpSendAgain,
+  paths.user.logOut,
+  paths.user.noAccess,
+  paths.user.forgottenPassword,
+  paths.user.passwordRequested,
+  paths.user.forgottenPasswordReset,
+  paths.inviteValidation.validateInvite,
+  paths.registerUser.registration,
+  paths.registerUser.subscribeService,
+  paths.registerUser.otpVerify,
+  paths.registerUser.reVerifyPhone,
+  paths.registerUser.logUserIn,
+  paths.selfCreateService.register,
+  paths.selfCreateService.confirm,
+  paths.selfCreateService.otpVerify,
+  paths.selfCreateService.otpResend,
+  paths.healthcheck.path,
+  paths.staticPaths.naxsiError
+]
+
+function flattenPaths(arrayThatMayContainObjects) {
+  return arrayThatMayContainObjects.reduce((paths, val) => {
+    if (typeof val === 'object' && val != null) {
+      return paths.concat(flattenPaths(Object.values(val)))
+    } else {
+      paths.push(val)
+      return paths
+    }
+  }, [])
+}
+
+describe('The Express router', () => {
+  it('should include authorisation middleware in stack for all paths not in exceptions', () => {
+    const app = {
+      use: sinon.spy(),
+      get: sinon.spy(),
+      post: sinon.spy(),
+      all: sinon.spy()
+    }
+
+    routes.bind(app)
+
+    // We currently call `app.use` to specify a list of paths to use certain middleware for, 
+    // including the authorisation middleware. We need to check that paths are either included in
+    // the array for this call, or the middleware is added to the stack individually for the route
+    const authenticatedPathsArg = app.use.getCalls()
+      .find(call => {
+        return Array.isArray(call.args[0]) &&
+          call.args.includes(oldAuthorisationMiddleware)
+      })
+      .args[0]
+    const authenticatedPaths = flattenPaths(authenticatedPathsArg)
```

Comment:
It took me a minute to fully get what was going on with this step before getting to the routes! I think this edge case coverage could also be removed if we're encouraging route based authorisation with the new auth middleware

Reply:
Yep, it can be removed down the line, maybe we should put a reminder in to do it on another story?